### ISSUE TITLE
Don't we need a return statement in forward method?

### ISSUE THREAD

**Issue by pranaymodukuru** [boathit/JuliaTorch#3]
I am new to Julia, but use Pytorch in python

I notice there is no return statement at the end of forward method, is it so in Julia? Please correct me if I am wrong

Just wanted to know how efficient is Calling Pytorch in Julia?
Is it fast enough?

**Comment by boathit** [comment]
> I notice there is no return statement at the end of forward method, is it so in Julia?

Yes, all expressions have default values in Julia and the `return` keyword is optional.

> Just wanted to know how efficient is Calling Pytorch in Julia?

In most cases, the overhead of calling PyTorch is quite low in Julia. The bottleneck lies in the GPU memory management, which we have to [release manually](https://github.com/boathit/JuliaTorch/blob/master/mlp.jl#L49) right now.

**Comment by pranaymodukuru** [comment]
I suppose we cannot slice index pytorch tensors from Julia as of yet..
May I know when or wether this option would be available?

**Comment by boathit** [comment]
Yes, that's correct. So we have to resort to the API like `index_select` by now.

**Comment by pranaymodukuru** [comment]
> Yes, that's correct. So we have to resort to the API like `index_select` by now.

Could you please give an example for that.

**Comment by boathit** [comment]
[Tensor selection](https://github.com/boathit/JuliaTorch/blob/master/maskedAutoRegressiveFlow.jl#L51) and [tensor assignment](https://github.com/boathit/JuliaTorch/blob/master/maskedAutoRegressiveFlow.jl#L57).

**Comment by pranaymodukuru** [comment]
I think these examples show selecting only a single value from a tensor, but I would like to select a slice of a tensor.

For example:
`a = torch.rand(1,2,28,28)`
 
`a[0,1, 3:5, 3:5]`

Gives me a small part of the 28x28 image let us say.

**Comment by boathit** [comment]
It seems we cannot slice a tensor along multiple dimensions simultaneously so far.
